oauth2-server/src/main/java/com/clouway/oauth2/AuthorizedClientActivity.java

```diff
@@ -21,5 +21,5 @@
    * @param instant the time of which client was requested access
    * @return the response
    */
-  Response execute(Client client, Identity identity, Set<String> scopes, Request request, DateTime instant);
+  Response execute(Client client, Identity identity, Set<String> scopes, Request request, DateTime instant, int index);
```

Comment:
You can pass all params with something more common like: ```Map<String, String> instead of just index. 

Whether a list of allowed parameters could be passed as parameter to ClientAuthorizationActivity to prevent re-passing of any parameter.